From 1c0ba6b53759ea57a6146b4ce4d902cfe5c44d3c Mon Sep 17 00:00:00 2001
From: Roland Fischer <nalor@gmx.net>
Date: Wed, 9 Dec 2015 21:50:43 +0100
Subject: [PATCH] Correct debug-log in _find_config_entry

'discid' sent to BD_DEBUG has not been converted to string before
and resulted in a broken output - corrected.
---
 src/libaacs/aacs.c |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/libaacs/aacs.c b/src/libaacs/aacs.c
index ae85612..60e678f 100644
--- a/src/libaacs/aacs.c
+++ b/src/libaacs/aacs.c
@@ -710,6 +710,7 @@ static void _find_config_entry(AACS *aacs, title_entry_list *ce,
                                uint8_t *mk, uint8_t *vuk)
 {
     char str[48];
+    char str2[48];
 
     aacs->uks = NULL;
     aacs->num_uks = 0;
@@ -731,7 +732,7 @@ static void _find_config_entry(AACS *aacs, title_entry_list *ce,
             hexstring_to_hex_array(mk, 16, ce->entry.mek);
 
             BD_DEBUG(DBG_AACS, "Found media key for %s: %s\n",
-                  ce->entry.discid, str_print_hex(str, mk, 16));
+                  str_print_hex(str2, ce->entry.discid, 20), str_print_hex(str, mk, 16));
         }
 
         if (ce->entry.vid) {
@@ -739,14 +740,14 @@ static void _find_config_entry(AACS *aacs, title_entry_list *ce,
                                     ce->entry.vid);
 
             BD_DEBUG(DBG_AACS, "Found volume id for %s: %s\n",
-                  ce->entry.discid, str_print_hex(str, aacs->vid, 16));
+                  str_print_hex(str2, ce->entry.discid, 20), str_print_hex(str, aacs->vid, 16));
         }
 
         if (ce->entry.vuk) {
             hexstring_to_hex_array(vuk, 16, ce->entry.vuk);
 
             BD_DEBUG(DBG_AACS, "Found volume unique key for %s: %s\n",
-                  ce->entry.discid, str_print_hex(str, vuk, 16));
+                  str_print_hex(str2, ce->entry.discid, 20), str_print_hex(str, vuk, 16));
         }
 
         if (ce->entry.uk) {
-- 
1.7.10.4

